[TECH] Ajouter un faux fournisseur d'identité pour faciliter les tests (PIX-10989)

lego-technix opened the pull request:
## :unicorn: Problème

Dans une étape prochaine, l'instanciation des services d'authentification pour les différents fournisseurs d’identité OIDC va faire un appel réseau à leur OpenID-configuration respective. On ne veut pas que chaque fois qu'un test est exécuté les différents fournisseurs d'identités OIDC soient sollicités. Et on ne veut pas non plus devoir mettre en place pour les tests  du stubbing/mocking pour chaque services d'authentification des différents fournisseurs d’identité OIDC.

## :robot: Proposition

Créer un faux fournisseur d’identité OidcExampleNet et l’utiliser dans les tests.

## :rainbow: Remarques

RAS

## :100: Pour tester

1. Vérifier que la CI passe
2. Vérifier qu’il n'y a pas de régression dans les SSO en testant une connexion et une déconnexion avec chacun des SSO

igorissen committed:
test(api): add createServerWithTestOidcProvider test tooling

Co-authored-by: Ismael Gorissen <ismael.gorissen@gmail.com>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: BKsensei <rebecca.kaci@gmail.com>

lego-technix committed:
test(api): reorder imports

Co-authored-by: Ismael Gorissen <ismael.gorissen@gmail.com>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: BKsensei <rebecca.kaci@gmail.com>

lego-technix committed:
test(api): adapt all tests to use new test OIDC Provider

Co-authored-by: Ismael Gorissen <ismael.gorissen@gmail.com>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: BKsensei <rebecca.kaci@gmail.com>

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr7973.review.pix.fr)
  * [App (.org)](https://app-pr7973.review.pix.org)
  * [Orga (.fr)](https://orga-pr7973.review.pix.fr)
  * [Orga (.org)](https://orga-pr7973.review.pix.org)
  * [Certif (.fr)](https://certif-pr7973.review.pix.fr)
  * [Certif (.org)](https://certif-pr7973.review.pix.org)
  * [1D](https://1d-pr7973.review.pix.fr)
  * [Admin](https://admin-pr7973.review.pix.fr)
  * [API](https://api-pr7973.review.pix.fr/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr7973.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr7973/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr7973/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr7973/environment)

igorissen committed:
feat(api): improve authentication service registry configurability

Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Rébecca Kaci <rebecca.kaci@gmail.com>

igorissen committed:
feat(api): add hapi configuration step

configure a list of OIDC provider services

Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Rébecca Kaci <rebecca.kaci@gmail.com>

lego-technix committed:
test(api): remove now useless truncate

All truncates are being taken care of in
api/db/database-builder/database-builder.js addListeners.

Co-authored-by: Ismael Gorissen <ismael.gorissen@gmail.com>
Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: BKsensei <rebecca.kaci@gmail.com>

lego-technix committed:
test(api): reorder exports

Co-authored-by: Ismael Gorissen <ismael.gorissen@gmail.com>
Co-authored-by: LEGO Technix <109212476+lego-technix@users.noreply.github.com>
Co-authored-by: Emmanuelle Bonnemay <emmanuelle.bonnemay@pix.fr>
Co-authored-by: BKsensei <rebecca.kaci@gmail.com>

igorissen reviewed (approved):
<img src="https://media3.giphy.com/media/KzKuco4UZwtC3gGnFK/giphy.gif"/>

lego-technix commented:
Testé fonctionnellement avec succès avec les SSO FWB, CNAV, Pays-de-la-Loire. Le SSO Pôle emploi était indisponible au moment des tests.

reibecca reviewed (approved):
![75655](https://github.com/1024pix/pix/assets/82950611/091d514c-994a-4ba0-a417-9455b8fc2505)
